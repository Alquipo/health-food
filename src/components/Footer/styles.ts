import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 10.3rem;
    padding: ${theme.spacings.small} 0;
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.xlarge};

    ${media.lessThan('medium')`
      flex-direction: column;
      justify-content: center;
      text-align: center;
      height: auto;
    `};
  `}
`

export const Copyright = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.secondary};

    ${media.lessThan('medium')`
    margin-top: ${theme.spacings.medium};
      order: 2;
    `};
  `}
`

export const Links = styled.div`
  ${({ theme }) => css`
    a {
      color: ${theme.colors.text};
      text-decoration: none;
      transition: ${theme.transition.fast};

      &:not(:first-child) {
        margin-left: ${theme.spacings.xsmall};
      }

      &:hover {
        text-decoration: underline;
      }

      ${media.lessThan('medium')`
        &:not(:first-child) {
          margin-top: ${theme.spacings.xxsmall};
         }
     `};
    }

    ${media.lessThan('medium')`
      order: 1;
      display: flex;
      flex-direction: column;
    `}
  `}
`
