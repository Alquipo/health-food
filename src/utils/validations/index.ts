import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
