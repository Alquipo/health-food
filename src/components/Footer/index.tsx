import Container from '../Container'
import * as S from './styles'

export default function Footer() {
  return (
    <Container>
      <S.Wrapper>
        <S.Copyright>Â© Copyrights 2019 Stack. All Rights Reserved.</S.Copyright>

        <S.Links>
          <a href="https://github.com/Alquipo">Privacy Policy</a>
          <a href="https://github.com/Alquipo">Terms and Conditions</a>
        </S.Links>
      </S.Wrapper>
    </Container>
  )
}
