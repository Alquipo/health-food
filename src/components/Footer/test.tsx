import { render, screen } from 'utils/test-utils'

import Footer from '.'

describe('<Footer/>', () => {
  it('should render a Footer Section correctly', () => {
    const { container } = render(<Footer />)

    expect(
      screen.getByText(/Â© Copyrights 2019 Stack. All Rights Reserved./i)
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /Privacy Policy/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /Terms and Conditions/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
