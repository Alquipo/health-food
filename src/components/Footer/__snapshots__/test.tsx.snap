// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render a Footer Section correctly 1`] = `
.c0 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 10.3rem;
  padding: 2.4rem 0;
  font-weight: 600;
  font-size: 2.0rem;
}

.c2 {
  color: #1D164D;
}

.c3 a {
  color: #9E9BAF;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c3 a:not(:first-child) {
  margin-left: 1.6rem;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    text-align: center;
    height: auto;
  }
}

@media (max-width:768px) {
  .c2 {
    margin-top: 3.2rem;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (max-width:768px) {
  .c3 a:not(:first-child) {
    margin-top: 0.8rem;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
  class="c0"
>
  <footer
    class="c1"
  >
    <p
      class="c2"
    >
      Â© Copyrights 2019 Stack. All Rights Reserved.
    </p>
    <div
      class="c3"
    >
      <a
        href="https://github.com/Alquipo"
      >
        Privacy Policy
      </a>
      <a
        href="https://github.com/Alquipo"
      >
        Terms and Conditions
      </a>
    </div>
  </footer>
</div>
`;
